components/script/dom/document.rs

```diff
@@ -376,8 +376,12 @@ impl Document {
     }
 
     // https://dom.spec.whatwg.org/#concept-document-url
-    pub fn url(&self) -> &Url {
-        &self.url
+    pub fn url(&self) -> Ref<Url> {
+        self.url.borrow()
```

Comment:
Well since the previously `document.url()` returns a reference, now change it to return the actual struct would results changes to a lot of places(roughly 50, and they need to be changed again if we implement refcounting later). If you can assure that is OK I will make the change.